File: frontend/console/src/api/signal-ai.js
import Axios from "@/http";

export function createSignalAi(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-ai', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSignalAi(signalAiId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/signal-ai/${ signalAiId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSignalAi(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/signal-ai', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalAi(signalAiId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/signal-ai/${ signalAiId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalAiWithFlag(signalAiId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/signal-ai/${ signalAiId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSignalAis() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/signal-ai/all').then(res => {
            resolve(res)
        })
    })
}

export function querySignalAi (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-ai/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/signal_ai.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SIGNAL_AI = "/config/signal-ai"
const URL_ROUTER_SIGNAL_AI_BY_ID = "/config/signal-ai/:signal-ai-id"
const URL_ROUTER_SIGNAL_AI_QUERY = "/config/signal-ai/query"
const URL_ROUTER_SIGNAL_AI_ALL = "/config/signal-ai/all"

func SignalAiRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SIGNAL_AI, config.CreateSignalAi)
	r.GET(URL_ROUTER_SIGNAL_AI_BY_ID, config.RetrieveSignalAi)
	r.PUT(URL_ROUTER_SIGNAL_AI, config.UpdateSignalAi)
	r.DELETE(URL_ROUTER_SIGNAL_AI_BY_ID, config.DeleteSignalAi)
	r.POST(URL_ROUTER_SIGNAL_AI_QUERY, config.QuerySignalAi)
	r.GET(URL_ROUTER_SIGNAL_AI_ALL, config.GetAllSignalAis)
	return r
}